Cargo.toml

```diff
@@ -50,6 +51,8 @@ async-std = {version = "1.12", optional = true}
 tokio = {version = "1.21", features = ["fs", "io-util"], optional = true, default-features = false}
 once_cell = "1.14"
 url = {version = "2.3", features = ["serde"]}
+async-trait = {version = "0.1.60", optional = true}
+futures = { version = "0.3.25", features = ["executor", "thread-pool"], optional = true}
```

Comment:
do we actually need the whole futures crate here?

Reply:
only for the spawn, but if we already depend always on either tokio or async-std then we can remove it.